webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/BinaryDataService.scala

```diff
@@ -22,7 +22,8 @@ class BinaryDataService(val dataBaseDir: Path,
                         val agglomerateServiceOpt: Option[AgglomerateService],
                         remoteSourceDescriptorServiceOpt: Option[RemoteSourceDescriptorService],
                         sharedChunkContentsCache: Option[AlfuCache[String, MultiArray]],
-                        datasetErrorLoggingService: Option[DatasetErrorLoggingService])(implicit ec: ExecutionContext)
+                        datasetErrorLoggingService: Option[DatasetErrorLoggingService],
+                        val remoteWKClient: Option[DSRemoteWebknossosClient])(implicit ec: ExecutionContext)
```

Comment:
is it really necessary for the binarydataservice to extend the datasetdeleter? I get the feeling it shouldn’t be involved in deletions. If not, this can also be removed again, and the client could be made non-optional in the deleter?

Reply:
Good idea!